src/altinn-app-frontend/src/features/form/layout/index.ts

```diff
@@ -99,6 +99,17 @@ export interface ILayoutCompDatePicker extends ILayoutCompBase<'DatePicker'> {
 }
 
 export type ILayoutCompDropdown = ILayoutCompBase<'Dropdown'> & ISelectionComponent;
+export interface ILayoutCompList extends ILayoutCompBase<'List'> {
+  tableHeaders?: string[];
+  sortableColumns?: string[];
+  pagination: IPagination;
+  dataList?: IDataList[];
+  dataListId?: string;
+  mapping?: IMapping;
+  secure?: boolean;
+  source?: IOptionSource;
+  fieldToStoreInDataModel: string;
```

Comment:
Not quite sure I agree with this one. I'd prefer to set up the component with `dataModelBindings` which is what we use for these things right now:
```
  "dataModelBindings": {
    "simpleBinding": "My.Model.Path"
  }
```
In this case we'd need a source -> target mapping, so I suggest trying to work with this format:
```
  "dataModelBindings": {
    "myTableColumn": "My.Model.Path",
    "case_number": "My.Model.CaseNumber"
  }
```
Where it would be possible to map multiple table columns in the chosen row to the data model.